cmake/alpakaCommon.cmake

```diff
@@ -343,6 +343,23 @@ if(NOT alpaka_HAS_STD_MATH_CONSTANTS)
     target_compile_definitions(alpaka INTERFACE "$<$<OR:$<CXX_COMPILER_ID:MSVC>,$<AND:$<COMPILE_LANGUAGE:CUDA>,$<PLATFORM_ID:Windows>>>:_USE_MATH_DEFINES>")
 endif()
 
+#-------------------------------------------------------------------------------
+# Include mdspan
+
+set(alpaka_USE_MDSPAN "INTERNAL" CACHE STRING "Use std::mdspan with alpaka")
+set_property(CACHE alpaka_USE_MDSPAN PROPERTY STRINGS "SYSTEM;INTERNAL;OFF")
+
+if (alpaka_USE_MDSPAN STREQUAL "SYSTEM")
+    find_package(mdspan REQUIRED)
+    target_link_libraries(alpaka INTERFACE std::mdspan)
+    target_compile_definitions(alpaka INTERFACE ALPAKA_USE_MDSPAN)
+endif()
+if (alpaka_USE_MDSPAN STREQUAL "INTERNAL")
```

Comment:
IMO this should be an `elseif()` and the case `OFF` should be handled too.
All other cases should be watched with `else()` to help the user to find typos e.g. `cmake -Dalpaka_USE_MDSPAN=INTERNAl`

Reply:
Done.